src/CoreWCF.ConfigurationManager/src/CoreWCF/Configuration/ExtendedProtectionPolicyElement.cs

```diff
@@ -0,0 +1,116 @@
+
+namespace CoreWCF.Configuration
+{
+    public sealed class ExtendedProtectionPolicyElement : ConfigurationElement
+    {
+        public ExtendedProtectionPolicyElement()
+        {
+            this.properties.Add(this.policyEnforcement);
+            this.properties.Add(this.protectionScenario);
+            this.properties.Add(this.customServiceNames);
+        }
+            
+        protected override ConfigurationPropertyCollection Properties
+        {
+            get { return this.properties; }
+        }
+
+        [ConfigurationProperty(ExtendedProtectionConfigurationStrings.PolicyEnforcement)]
+        public PolicyEnforcement PolicyEnforcement
+        {
+            get
+            {
+                return (PolicyEnforcement)this[this.policyEnforcement];
+            }
+            set
+            {
+                this[this.policyEnforcement] = value;
+            }
+        }
+
+        [ConfigurationProperty(ExtendedProtectionConfigurationStrings.ProtectionScenario, DefaultValue = ProtectionScenario.TransportSelected)]
+        public ProtectionScenario ProtectionScenario
+        {
+            get
+            {
+                return (ProtectionScenario)this[this.protectionScenario];
+            }
+            set
+            {
+                this[this.protectionScenario] = value;
+            }
+        }
+
+        [ConfigurationProperty(ExtendedProtectionConfigurationStrings.CustomServiceNames)]
+        public ServiceNameElementCollection CustomServiceNames
+        {
+            get
+            {
+                return (ServiceNameElementCollection)this[this.customServiceNames];
+            }
+        }
+
+        public ExtendedProtectionPolicy BuildPolicy()
+        {
+            if (PolicyEnforcement == PolicyEnforcement.Never)
+            {
+                return new ExtendedProtectionPolicy(PolicyEnforcement.Never);
+            }
+
+            ServiceNameCollection spns = null;
+
+            ServiceNameElementCollection spnCollection = CustomServiceNames;
+            if (spnCollection != null && spnCollection.Count > 0)
+            {
+                List<string> spnList = new List<string>(spnCollection.Count);
+                foreach (ServiceNameElement element in spnCollection)
+                {
+                    spnList.Add(element.Name);
+                }
+                spns = new ServiceNameCollection(spnList);
+            }
+
+            return new ExtendedProtectionPolicy(PolicyEnforcement, ProtectionScenario, spns);
+        }
+
+        ConfigurationPropertyCollection properties = new ConfigurationPropertyCollection();
+
+        private static PolicyEnforcement DefaultPolicyEnforcement
+        {
+            get
+            {
+                return PolicyEnforcement.Never;
+            }
+        }
+
+        readonly ConfigurationProperty policyEnforcement =
+            new ConfigurationProperty(ExtendedProtectionConfigurationStrings.PolicyEnforcement,
```

Comment:
Missing the `private` accessibility modifier. Also naming convention is that private fields start with an underscore.